pkg/push/api/api.go

```diff
@@ -561,28 +569,152 @@ func (s *PushService) UpdatePush(
 		}
 		return nil, dt.Err()
 	}
-	return &pushproto.UpdatePushResponse{}, nil
+
+	if updatedPushPb != nil {
+		// For security reasons we remove the service account from the API response
+		updatedPushPb.FcmServiceAccount = ""
+	}
+	return &pushproto.UpdatePushResponse{
+		Push: updatedPushPb,
+	}, nil
 }
 
-func (s *PushService) validateUpdatePushRequest(
+func (s *PushService) updatePushNoCommand(
 	ctx context.Context,
 	req *pushproto.UpdatePushRequest,
 	localizer locale.Localizer,
-) error {
-	if req.Id == "" {
-		dt, err := statusIDRequired.WithDetails(&errdetails.LocalizedMessage{
+	editor *eventproto.Editor,
+) (*pushproto.UpdatePushResponse, error) {
+	if err := s.validateUpdatePushRequestNoCommand(ctx, req, localizer); err != nil {
+		return nil, err
+	}
+	var updatedPushPb *pushproto.Push
+	var updatePushTagsEvent *eventproto.Event
+	var updatePushNameEvent *eventproto.Event
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
 			Locale:  localizer.GetLocale(),
-			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "id"),
+			Message: localizer.MustLocalize(locale.InternalServerError),
 		})
 		if err != nil {
-			return statusInternal.Err()
+			return nil, statusInternal.Err()
 		}
-		return dt.Err()
+		return nil, dt.Err()
 	}
-	if s.isNoUpdatePushCommand(req) {
-		dt, err := statusNoCommand.WithDetails(&errdetails.LocalizedMessage{
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		pushStorage := v2ps.NewPushStorage(tx)
+		push, err := pushStorage.GetPush(ctx, req.Id, req.EnvironmentNamespace)
+		if err != nil {
+			return err
+		}
+		prev := &domain.Push{}
+		if err = copier.Copy(prev, push); err != nil {
+			return err
+		}
+
+		if req.Name != nil {
+			push.Name = req.Name.Value
+			updatePushNameEvent, err = domainevent.NewEvent(
+				editor,
+				eventproto.Event_PUSH,
+				push.Id,
+				eventproto.Event_PUSH_RENAMED,
+				&eventproto.PushRenamedEvent{
+					Name: req.Name.Value,
+				},
+				req.EnvironmentNamespace,
+				push,
+				prev,
+			)
+			if err != nil {
+				return err
+			}
```

Comment:
As I commented before, the business logic should be in the domain layer to maintain consistency.

Reference:
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/domain/feature.go#L957-L1043

Reply:
I have moved to the domain layer.